From e93f0037e5be48941caf1a35ff5c088a826d4844 Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Tue, 27 May 2025 00:08:02 -0400
Subject: [PATCH 065/378] winebus: Ignore steam input virtual controller when
 PROTON_PREFER_SDL is set.

---
 dlls/winebus.sys/bus_udev.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/dlls/winebus.sys/bus_udev.c b/dlls/winebus.sys/bus_udev.c
index 2aba9b91f7b..5eabf42add0 100644
--- a/dlls/winebus.sys/bus_udev.c
+++ b/dlls/winebus.sys/bus_udev.c
@@ -1739,7 +1739,16 @@ static void udev_add_device(struct udev_device *dev, int fd)
     }
 
     if (desc.vid == 0x28de && desc.pid == 0x11ff && !strcmp(subsystem, "input"))
+    {
+        char *env = getenv("PROTON_NO_STEAMINPUT");
+        if (env && !strcmp(env, "1"))
+        {
+            TRACE("evdev %s: gnoring steam input virtual controller\n", debugstr_a(devnode));
+            close(fd);
+            return;
+        }
         TRACE("evdev %s: detected steam input virtual controller\n", debugstr_a(devnode));
+    }
     else if (is_sdl_ignored_device(desc.vid, desc.pid))
     {
         TRACE("evdev %s: ignoring %s, in SDL ignore list\n", debugstr_a(devnode), debugstr_device_desc(&desc));
-- 
2.51.0

